### ISSUE TITLE
Too EASY to DELETE ALL database

### ISSUE THREAD

**Issue by IRediTOTO** [prisma/prisma#9114]
## Problem

Hi, today I tried to run `npx prisma db push` 
I got this warning
`√ To apply this change we need to reset the database, do you want to continue? All data will be lost. ... yes`
Yes, I didn't read careful, I think it just deletes data in one table which data can be deleted. Just press `y` and all deleted
And yes, my fool. But can you make it  harder to delete all the data?
## Suggested solution

Add one more step if you reset all the data User must input `DELETE ALL` to delete all data

**Comment by danieljaguiar** [comment]
Same here. I was a blink away from deleting the whole thing.

Can we get a parameter `--allowReset`?

> To apply this change we need to reset the database. If you want to do so, run the command again with --allowReset

**Comment by tylerjbainbridge** [comment]
Bumping this, it's far too easy to delete your entire database in one keypress! There should be a flag to prevent this.

I get scared every time I run a migration because of this prompt.

**Comment by dzaulochnov** [comment]
Main question is why it erases data it all tables and not in only changed.

**Comment by janpio** [comment]
Probably because we considered this a "out of the ordinary" situation where using the big hammer is better than trying to do small course corrections. But it's a good idea and suggestion, you should open a feature request so we can officially consider it @dzaulochnov. Thanks.

**Comment by dzaulochnov** [comment]
> Probably because we considered this a "out of the ordinary" situation where using the big hammer is better than trying to do small course corrections. But it's a good idea and suggestion, you should open a feature request so we can officially consider it @dzaulochnov. Thanks.

Thanks for fast answer! Probably, will do it.

**Comment by tylerjbainbridge** [comment]
A decent solution might be having to type a phrase vs just one keypress.

> Type "confirm" to reset database .... etc

**Comment by janpio** [comment]
> ... having to type a phrase vs just one keypress.

That is a good suggestion. 

Here are all the places (that I could find) that need to be adapted to cover all the cases where we do dangerous things with one key press:
https://github.com/prisma/prisma/blob/99b1ddcad2e4826371c1c9f9f5fb304be945488d/packages/migrate/src/commands/DbPush.ts#L169
https://github.com/prisma/prisma/blob/99b1ddcad2e4826371c1c9f9f5fb304be945488d/packages/migrate/src/commands/DbPush.ts#L219
https://github.com/prisma/prisma/blob/99b1ddcad2e4826371c1c9f9f5fb304be945488d/packages/migrate/src/commands/MigrateDev.ts#L380
https://github.com/prisma/prisma/blob/99b1ddcad2e4826371c1c9f9f5fb304be945488d/packages/migrate/src/commands/MigrateReset.ts#L96

Instead we want to require input of a specific command: https://github.com/prisma/prisma/blob/99b1ddcad2e4826371c1c9f9f5fb304be945488d/packages/migrate/src/commands/DbDrop.ts#L100

Note that this will be a breaking change and any implementation has to wait until the next major Prisma version, Prisma 6.

**Comment by tylerjbainbridge** [comment]
This would be great, @janpio!

**Comment by danielvoelk** [comment]
Would be a great feature

**Comment by hxgdzyuyi** [comment]
I suggest that users should be prompted to drop the database themselves. It's too dangerous.

**Comment by joelrb** [comment]
Agree, only affected tables should be reset not the entire db. Also should allow reset db only in development and not production.
